.circleci/config.yml

```diff
@@ -1,182 +1,482 @@
+          If you provide a direct list of files, e.g.
+            "echo 'spec/one_spec.rb spec/two_spec.rb spec/red_spec.rb spec/blue_spec.rb'"
+           And you wish to maintain proper parallelism support, you can pipe it to "circleci tests split":
+            "$(echo 'spec/one_spec.rb \n spec/two_spec.rb \n spec/red_spec.rb \n spec/blu_spec.rb' | circleci tests split --split-by=timings)"
+          NOTE: 'circleci tests split' wants a newline separated list:
+              spec/seuss/one_spec.rb
+              spec/seuss/two_spec.rb
+              spec/seuss/red_spec.rb
+              spec/seuss/blu_spec.rb
+          Don't forget to use a subshell.
+      profile-specs:
+        description: |
+          State the number of specs you wish to evaluate for slowness using RSpec profiler. Default is 0.
+        type: integer
+        default: 0
+    steps:
       - run:
-          name: Wait for ElasticSearch
-          command: dockerize -wait http://127.0.0.1:9200 -timeout 1m
+          name: Run RSpec Tests
+          environment:
+            OUT_PATH: tmp/test-results
+            PROFILE_COUNT: "<< parameters.profile-specs >>"
+          command: |
+            shopt -s globstar
+            mkdir -p $OUT_PATH
+            TEST_PATTERN=<< parameters.test_pattern >>
+            TEST_FILES=<< parameters.test_files >>
+            RSPEC_COMMAND="bundle exec rspec --profile $PROFILE_COUNT --format RspecJunitFormatter --out $OUT_PATH/results.xml --format progress --order defined $TEST_FILES"
+            printf "Executing specs with the following command:\n\n"
+            echo ${RSPEC_COMMAND}
+            printf "\n"
+            ${RSPEC_COMMAND}
+            echo "RSpec exited with code $?"
       - run:
-          name: Run Tests
-          command: bundle exec rspec
-      - persist_to_workspace:
-          root: .
-          paths:
-            - vendor/bundle
+          name: Double check RSpec Failures
+          environment:
+            OUT_PATH: tmp/test-results
+          command: |
+            FILE=$OUT_PATH/results.xml
+            if [ -f $FILE ]; then
+              FAIL_COUNT=$(ruby -r rexml/document -e 'include REXML; puts XPath.first(Document.new($stdin), "string(//testsuite/@failures)")' < $FILE )
+              if [ $FAIL_COUNT -gt 0 ]; then
+                printf "IF YOU ARE READING THIS, TESTS FAILED BUT \"rspec\" RETURNED A 0 EXIT CODE AND LOOKS GREEN!\n\n"
+                printf "PLEASE LOOK AT THE PREVIOUS STEP TITLED \"Run RSpec Tests\" TO SEE SPEC FAILURE INFO!\n\n"
+                printf "You may also look at the \"TESTS\" tab, which will show failing spec info.\n\n"
+                printf "Additionally, $FILE is available for download and inspection in the \"ARTIFACTS\" tab.\n\n"
+                exit 1
+              fi
+            else
+              echo "$FILE not found..."
+              exit 0
+            fi
+            echo "No uncaught RSpec failures detected!"
+
+  install_gem_version:
```

Comment:
This isn't being used but left it in here in case anyone wanted to test with activemodel/activerecord/activesupport/etc gem versions.

Example: https://github.com/doximity/track_ballast/blob/master/.circleci/config.yml#L236